### PR TITLE
Enable the status subresource for build-pipeline CRDs

### PR DESCRIPTION
**Pull request by dprotaso**
Fixes: #364

This enables the status subresource in order for `metadata.generation` to start incrementing. It requires Kubernetes 1.11+

### CODE REVIEW

File: config/200-clusterrole.yaml [hunk 1]
```diff
@@ -12,12 +12,12 @@ rules:
   - apiGroups: ["admissionregistration.k8s.io"]
     resources: ["mutatingwebhookconfigurations"]
     verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
-  - apiGroups: ["apiextensions.k8s.io"]
-    resources: ["customresourcedefinitions"]
-    verbs: ["get", "list", "create", "update", "delete", "patch", "watch"]
```
Comment (dprotaso) [inline config/200-clusterrole.yaml]: FYI - I removed this because I don't expect a controller to create custom resource definitions

**Comment by bobcatfish** [comment]
Thanks for working on this @dprotaso :tada: !

Moving over my feedback from the issue (#364), could you add some more detail to the commit messages about why these changes are needed?

From https://github.com/knative/build/pull/516 I can see this comment that explains it a bit:

> We enable the status subresource on the templates CRDs so metadata.generation bumping will work in 1.11

But I am missing the context of what the status subresource is, why it's needed, what the metadata.generation bumping is about (maybe a link would help there?)

And it would really be useful to have all this in the commit messages (see our [commit message standard](https://github.com/knative/build-pipeline/blob/master/CONTRIBUTING.md#commit-messages))

The current commit message "remove spec.generation from build-pipeline types" I think is missing a lot of context

**Comment by bobcatfish** [comment]
```
I0108 15:52:23.904] Running tests with 'go test -race -v  -count=1 -tags=e2e -timeout=20m ./test '
W0108 15:52:58.301] # github.com/knative/build-pipeline/test [github.com/knative/build-pipeline/test.test]
W0108 15:52:58.302] test/helm_task_test.go:277:4: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
W0108 15:52:58.302] test/pipelinerun_test.go:466:4: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
W0108 15:52:58.303] test/pipelinerun_test.go:593:4: unknown field 'PipelineTrigger' in struct literal of type "github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1".PipelineRunSpec
```

oh dear, i think this is 1. legit 2. my fault from #366 and maybe not caught in the PR due to #361 I will look into this!

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish this PR is still in progress, I checked with @dprotaso and he will update it soon after he gets some changes in `pkg`

**Comment by bobcatfish** [comment]
it's happening

lol okay you got me @dprotaso , im gonna give up my commit message crusade in this one case XD gg

/lgtm
/meow space

**Pull request merged**